packages/venia-ui/lib/components/CartPage/GiftCards/giftCards.js

```diff
@@ -0,0 +1,178 @@
+
+const errorIcon = <Icon src={AlertCircleIcon} attrs={{ width: 18 }} />;
+
+const GiftCards = props => {
+    const talonProps = useGiftCards({
+        applyCardMutation: APPLY_GIFT_CARD_MUTATION,
+        cardBalanceQuery: GET_GIFT_CARD_BALANCE_QUERY,
+        cartQuery: GET_CART_GIFT_CARDS_QUERY,
+        removeCardMutation: REMOVE_GIFT_CARD_MUTATION
+    });
+    const {
+        applyGiftCard,
+        canTogglePromptState,
+        checkBalanceData,
+        checkGiftCardBalance,
+        errorLoadingGiftCards,
+        errorRemovingCard,
+        giftCardsData,
+        isLoadingGiftCards,
+        isApplyingCard,
+        isCheckingBalance,
+        isRemovingCard,
+        removeGiftCard,
+        setFormApi,
+        shouldDisplayCardBalance,
+        shouldDisplayCardEntry,
+        shouldDisplayCardError,
+        submitForm,
+        togglePromptState
+    } = talonProps;
+
+    const [, { addToast }] = useToasts();
+    useEffect(() => {
+        if (errorRemovingCard) {
+            addToast({
+                type: 'error',
+                icon: errorIcon,
+                message: 'Unable to remove gift card. Please try again.',
+                dismissable: true,
+                timeout: 7000
+            });
+        }
+    }, [addToast, errorRemovingCard]);
+
+    if (isLoadingGiftCards) {
+        return <LoadingIndicator>{`Loading Gift Cards...`}</LoadingIndicator>;
+    }
+    if (errorLoadingGiftCards) {
+        return (
+            <span>
+                {`There was an error loading gift cards. Please try again.`}
+            </span>
+        );
+    }
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const cardEntryErrorMessage = shouldDisplayCardError
+        ? `Invalid card. Please try again.`
+        : null;
+
+    let appliedGiftCards = null;
+    if (giftCardsData.length > 0) {
+        const cardList = giftCardsData.map(giftCardData => {
+            const { code } = giftCardData;
+
+            return (
+                <GiftCard
+                    code={code}
+                    removeGiftCard={removeGiftCard}
+                    isRemovingCard={isRemovingCard}
+                    key={code}
+                />
+            );
+        });
+
+        appliedGiftCards = (
+            <div className={classes.cards_container}>{cardList}</div>
+        );
+    }
+
+    const addCardContents = (
+        <button className={classes.show_entry} onClick={togglePromptState}>
+            {`+ Add another gift card`}
+        </button>
+    );
+    const cardEntryContents = (
+        <div className={classes.card}>
+            <Field
+                classes={{ root: classes.entry }}
+                id={classes.card}
+                label="Gift Card Number"
+            >
+                <div className={classes.card_input_container}>
+                    <TextInput
+                        id={classes.card}
+                        classes={{ icons: classes.card_input }}
```

Comment:
I think this is leftover, right?

Reply:
Yes, thanks! 

Resolved in https://github.com/magento/pwa-studio/pull/2124/commits/67a1061ae9e205d3f676c3b035c5cd3048dcc8e0.